rosetta-source/src/main/rosetta/event-position-type.rosetta

```diff
@@ -38,6 +44,49 @@ type AggregationParameters: <" Parameters to be used to filter events that are r
+    party Party (0..*) <"Defines all parties involved for the list of inventory records in this set of inventory. For example, when used to describe securities lending availability, this could hold the sender of the availability, the intended recipient, the beneficial owner(s), the lender (which may differ from the sender as the lender may have the same piece of availability going through multiple agents), an agent or a venue.">
+    partyRole PartyRole (0..*) <"Defines the role(s) that party(ies) may have in relation to the inventory.">
+    availableInventoryRecord AvailableInventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+	
+    condition ValidPartyRole: <"Restrict roles to be only those that would be associated to the overall list of availability">
+        if partyRole exists then
+            partyRole -> role any = PartyRoleEnum -> AgentLender 
+            or partyRole -> role any = PartyRoleEnum -> BeneficialOwner            
+            or partyRole -> role any = PartyRoleEnum -> Borrower
+            or partyRole -> role any = PartyRoleEnum -> Custodian
+            or partyRole -> role any = PartyRoleEnum -> Lender
+
+type AvailableInventoryRecord extends InventoryRecord: <"An individual piece of available inventory. This represents a single security and its associated criteria. The criteria are used to describe any restrictions on the securities.">
+    expirationDateTime zonedDateTime (0..1) <"There may be a set period/time restriction associated to the security.">
+    collateral CollateralProvisions (0..*) <"The type of collateral can often be required when determining if the piece of availability being described is suitable for a party.">
+    partyRole PartyRole (0..*) <"An individual security may be held by several agents. Including the party role at this level allows us to reference the party holding this specific item.">
+    availableQuantity Quantity (0..1) <"The quantity of the security that is available"> 
+    interestRate Price (0..1) <"An optional element which can be used to hold a rate associated to this piece of availability.">
+
+    condition InterestRate: <"Where an Interest Rate has been specified then the Price Type must be set to interest rate">
+        if interestRate exists then
+            interestRate -> priceType = PriceTypeEnum -> InterestRate
+
+    condition ValidPartyRole: <"Restrict roles to be only those that would be associated to an individual piece of availability">
+        if partyRole exists then
+            partyRole -> role any = PartyRoleEnum -> AgentLender 
+            or partyRole -> role any = PartyRoleEnum -> BeneficialOwner            
+            or partyRole -> role any = PartyRoleEnum -> Custodian
+            or partyRole -> role any = PartyRoleEnum -> Lender
```

Comment:
The condition as written doesn't achieve the objective, because you're only testing that `any` is in the allowed list. So if 1 is in and 1 is out, the condition still passes.

You actually need to write (with only 2 enum values for simplicity):

```
condition ValidPartyRole:
        partyRole
            extract ( role = PartyRoleEnum -> Lender or role = PartyRoleEnum -> Custodian )
            then all = True
```

Which, if you have more allowable values, could be written more compactly as:

```
condition ValidPartyRole:
        partyRole
            extract ([ PartyRoleEnum -> Lender, PartyRoleEnum -> Custodian ] contains role)
            then all = True
```

Reply:
Thanks for the tip Leo! I have updated the conditions as suggested.